ocaml/idl/datamodel_repository.ml

```diff
@@ -0,0 +1,82 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let lifecycle = [Published, rel_next, ""]
+
+let introduce = call
+    ~name:"introduce"
+    ~in_oss_since:None
+    ~in_product_since:rel_next
+    ~doc:"Create a new repository record in the database only"
+    ~params:[
+      String, "name_label", "The name of the repository";
+      String, "binary_url", "Base URL of binary packages in this repository";
+      String, "source_url", "Base URL of source pacakges in this repository";
+    ]
+    ~result:(Ref _repository, "The ref of the created repository record.")
+    ~allowed_roles:_R_POOL_OP
+    ()
+
+let forget = call
+    ~name:"forget"
+    ~in_oss_since:None
+    ~in_product_since:rel_next
+    ~doc:"Remove a repository record from the database"
```

Comment:
`Remove the repository from the database` - the repo in the argument will be removed so it is better to use the definite article (the)